components/EventBanner.tsx

```diff
@@ -1,25 +1,25 @@
 // CSS
 import styles from '../styles/components/EventBanner.module.scss';
 // Config
-import * as config from '../config.yaml';
+import config from '../config.yaml';
 import ReactMarkdown from 'react-markdown';
 
 interface Props {
   color?: string;
-  contextKey?: string;
+  contextKey?: keyof typeof config;
 }
 
 export default function EventBanner({
   color = 'event-banner',
   contextKey = 'bannerInfo',
 }: Props) {
-  const bannerInfo = config[contextKey] || [];
+  const bannerInfo = config[contextKey];
 
-  if (bannerInfo.length === 0 || bannerInfo?.hidden) {
+  if (!bannerInfo || (Array.isArray(bannerInfo) && bannerInfo.length === 0) || (bannerInfo as any)?.hidden) {
```

Comment:
The `as any` is not the way todo this can you add a todo linking to this issue like `//TODO(#12): Improve yaml typechecking`.